### PR TITLE
feat: Add manifest file.

### PR DESCRIPTION
**Pull request by abhidixi11**
Add manifest file to C-SDK repository

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the MIT license.

### CODE REVIEW

File: manifest.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+name : "AWS_IoT_Device_SDK_for_Embedded_C"
```
Comment (leegeth) [inline manifest.yml:1]: How is this manifest file going to be used?

File: manifest.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+version: "202009.00"                              
+description: |-                                  
+  "The AWS IoT Device SDK for Embedded C is a collection of C source 
+  files that can be used in embedded applications to securely connect 
+  to AWS IoT Core and interact with other AWS IoT services on AWS Cloud. \n
+  \n
+  See dependencies for included libraries."         
+dependencies:                                       
+  - name: "coreMQTT"                                
+    version: "1.0.0"
```
Comment (leegeth) [inline manifest.yml:11]: is it 1.0.0 or v1.0.0?

File: manifest.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    version: "1.0.0"                                
+    repository:                                     
+      type: "git"                                  
+      url: "https://github.com/aws/Device-Shadow-for-AWS-IoT-embedded-sdk"
+  - name: "CMock"                                
+    version: "2.5.2"                             
+    repository:
+      type: "git"                
+      url:  "https://github.com/ThrowTheSwitch/CMock"
+license: "MIT"
```
Comment (leegeth) [inline manifest.yml:35]: better to add a new line to the end of file.

File: manifest.yml (hunk 2 above)
Comment (abhidixi11) [reply manifest.yml:11]: That is for PMs to review.

File: manifest.yml (hunk 3 above)
Comment (abhidixi11) [reply manifest.yml:35]: OK. Will do.

File: manifest.yml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  files that can be used in embedded applications to securely connect 
+  to AWS IoT Core and interact with other AWS IoT services on AWS Cloud. \n
+  \n
+  See dependencies for included libraries."
+dependencies:
+  - name: "coreMQTT"
+    version: "1.0.0"
+      type: "git"
+      url: "https://github.com/FreeRTOS/coreMQTT"
+  - name: "coreHTTP"
```
Comment (jainamt) [inline manifest.yml:14]: http is not getting released now.

File: manifest.yml [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    version: "1.0.0" 
+    repository:
+      type: "git"
+      url: "https://github.com/FreeRTOS/coreJSON"
+  - name: "Device-Shadow-for-AWS-IoT-embedded-sdk" 
+    version: "1.0.0"                                
+    repository:                                     
+      type: "git"                                  
+      url: "https://github.com/aws/Device-Shadow-for-AWS-IoT-embedded-sdk"
+  - name: "CMock"
```
Comment (jainamt) [inline manifest.yml:29]: CMock is needed only for unit tests. I dont think we should list it.

Comment (abhidixi11) [reply manifest.yml:29]: I had discussion with Rashed earlier and he feels it should be there. By that logic we also need CBMC repo as dependency and Rashed agrees adding that. So we either remove CMock or add CBMC related dependency.

File: manifest.yml (hunk 4 above)
Comment (abhidixi11) [reply manifest.yml:14]: This pull request is for master, the one for release candidate does not have HTTP.

**Comment by abhidixi11** [comment]
Closing this PR as we have decided to add manifest file for RC branch ( and eventually tag )only  for this release. We will revisit master branch manifest file later.

**Pull request closed without merging**